core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/organism/AlertDialog.kt

```diff
@@ -0,0 +1,223 @@
+
+package app.k9mail.core.ui.compose.designsystem.organism
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import app.k9mail.core.ui.compose.common.LightDarkPreviews
+import app.k9mail.core.ui.compose.designsystem.atom.button.ButtonText
+import app.k9mail.core.ui.compose.designsystem.atom.icon.IconError
+import app.k9mail.core.ui.compose.designsystem.atom.icon.IconInfo
+import app.k9mail.core.ui.compose.designsystem.atom.icon.IconSuccess
+import app.k9mail.core.ui.compose.designsystem.atom.icon.IconWarning
+import app.k9mail.core.ui.compose.designsystem.atom.text.TextBody2
+import app.k9mail.core.ui.compose.designsystem.atom.text.TextSubtitle1
+import app.k9mail.core.ui.compose.theme.K9Theme
+import app.k9mail.core.ui.compose.theme.MainTheme
+import androidx.compose.material.AlertDialog as MaterialAlertDialog
+
+@Composable
+fun AlertDialog(
+    title: String,
+    text: String,
+    confirmButtonText: String,
+    onConfirmButtonClick: () -> Unit,
+    onDismissRequest: () -> Unit,
+    modifier: Modifier = Modifier,
+    type: AlertDialogType = AlertDialogType.Info,
+    hasTitleIcon: Boolean = false,
+    dismissButtonText: String? = null,
+    onDismissButtonClick: () -> Unit = {},
+) {
+    MaterialAlertDialog(
+        onDismissRequest = onDismissRequest,
+        confirmButton = configureConfirmButton(confirmButtonText, onConfirmButtonClick),
+        modifier = modifier,
+        dismissButton = configureDismissButton(dismissButtonText, onDismissButtonClick),
+        title = configureTitle(title, type, hasTitleIcon),
+        text = configureText(text),
+    )
+}
+
+private fun configureTitle(title: String?, type: AlertDialogType, hasIcon: Boolean): @Composable (() -> Unit)? {
+    return if (title != null) {
+        {
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.Start,
+            ) {
+                ConfigureTitleIcon(hasIcon, type)
+                TextSubtitle1(text = title)
+            }
+        }
+    } else {
+        null
+    }
+}
+
+@Composable
+private fun ConfigureTitleIcon(
+    hasIcon: Boolean,
+    type: AlertDialogType,
+) {
+    if (hasIcon) {
+        val modifier = Modifier.padding(end = MainTheme.spacings.default)
+        when (type) {
+            AlertDialogType.Success -> IconSuccess(modifier = modifier)
+            AlertDialogType.Error -> IconError(modifier = modifier)
+            AlertDialogType.Warning -> IconWarning(modifier = modifier)
+            AlertDialogType.Info -> IconInfo(modifier = modifier)
+        }
+    }
+}
+
+private fun configureText(text: String?): @Composable (() -> Unit)? {
+    return if (text != null) {
```

Comment:
`AlertDialog` currently doesn't allow `text` to be `null`. We could change the `AlertDialog` signature or remove this check.